plugins/woocommerce/src/Blocks/BlockTypes/Reviews/ProductReviews.php

```diff
@@ -48,6 +31,12 @@ protected function render( $attributes, $content, $block ) {
 		if ( ! comments_open() ) {
 			return '';
 		}
-		return $content;
+
+		$p = new \WP_HTML_Tag_Processor( $content );
+		$p->next_tag( 'div' );
+		$p->set_attribute( 'data-wp-interactive', $this->get_full_block_name() );
+		$p->set_attribute( 'data-wp-router-region', $this->get_full_block_name() );
```

Comment:
Not a strong opinion (and I'm not even sure if it would work), but I feel it would be safer to add these attributes in a wrapper `<div>` than modifying the block's content. While it has the drawback that it adds an extra `<div>`, it makes it resilient to any changes in the block's markup.

Reply:
It will work, but we don't have to be extra cautious here IMO. The markup of the Product Reviews block is the contract that needs to be held; it's only the wrapper `div` in this case.

By injecting the directives into the wrapper div, we can match the markup between the editor and the frontend, making styling easier. If we want to change the markup of the block, the logic here, which is a part of the block rendering, should be updated as well.

The markup might be changed for sure, but it's not likely to change here, and we're injecting the directive into the wrapper `div` element, which is even less likely to change.